src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -140,8 +140,8 @@ public void start(final Map<String, String> props) {
 
                 // If the cursor is invalidated add the record and return calls
                 if (changeStreamDocument.getString("operationType", new BsonString("")).getValue().equalsIgnoreCase(INVALIDATE)) {
-                    LOGGER.info("Cursor has been invalidated, no further messages will be published");
-                    isRunning.set(false);
+                    LOGGER.info("Cursor has been invalidated.");
+                    cursor = null;
```

Comment:
Not sure if we should assume that the cursor is exhausted.  Consider calling `close` first.

Reply:
Added

Reply:
I reverted this change as Invalidate events already close the cursor.  https://docs.mongodb.com/manual/reference/change-events/#invalidate-event

Reply:
Ack